PWGHF/D2H/Tasks/taskSigmacToCascade.cxx

```diff
@@ -0,0 +1,205 @@
+    registry.add("Data/hYSc0PlusPlusVsPt", "#Sigma_{c}^{0, ++} candidates; y (rapidity); #it{p}_{T}(#Sigma_{c}^{0}) (GeV/#it{c});", {HistType::kTH2F, {axisY, axisPt}});                                                                 /// Σc0,++
+    registry.add("Data/hPhiSc0", "#Sigma_{c}^{0} candidates; #Phi ; counts;", {HistType::kTH1F, {axisPhi}});                                                                                                                             /// Σc0
+    registry.add("Data/hPhiScPlusPlus", "#Sigma_{c}^{++} candidates; #Phi ; counts;", {HistType::kTH1F, {axisPhi}});                                                                                                                     /// Σc++
+    registry.add("Data/hPhiSc0PlusPlus", "#Sigma_{c}^{0, ++} candidates; #Phi ; counts;", {HistType::kTH1F, {axisPhi}});                                                                                                                 /// Σc0,++
+
+    /// softPions
+    registry.add("Data/hPtSoftPiSc0", "#pi^{-} #leftarrow #Sigma_{c}^{0} candidates; #it{p}_{T}(#pi^{-} #leftarrow #Sigma_{c}^{0}) (GeV/#it{c}); counts;", {HistType::kTH1F, {axisPt}});                 /// Σc0
+    registry.add("Data/hPtSoftPiScPlusPlus", "#pi^{+} #leftarrow #Sigma_{c}^{++} candidates; #it{p}_{T}(#pi^{+} #leftarrow #Sigma_{c}^{0}) (GeV/#it{c}); counts;", {HistType::kTH1F, {axisPt}});         /// Σc++
+    registry.add("Data/hPtSoftPiSc0PlusPlus", "#pi^{#pm} #leftarrow #Sigma_{c}^{0, ++} candidates; #it{p}_{T}(#pi^{#pm} #leftarrow #Sigma_{c}^{0}) (GeV/#it{c}); counts;", {HistType::kTH1F, {axisPt}}); /// Σc0,++
+    registry.add("Data/hEtaSoftPiSc0", "#pi^{-} #leftarrow #Sigma_{c}^{0} candidates; #eta ; counts;", {HistType::kTH1F, {axisEta}});                                                                    /// Σc0
+    registry.add("Data/hEtaSoftPiScPlusPlus", "#pi^{+} #leftarrow #Sigma_{c}^{++} candidates; #eta ; counts;", {HistType::kTH1F, {axisEta}});                                                            /// Σc++
+    registry.add("Data/hEtaSoftPiSc0PlusPlus", "#pi^{#pm} #leftarrow #Sigma_{c}^{0, ++} candidates; #eta ; counts;", {HistType::kTH1F, {axisEta}});                                                      /// Σc0,++
+    registry.add("Data/hPhiSoftPiSc0", "#pi^{-} #leftarrow #Sigma_{c}^{0} candidates; #Phi ; counts;", {HistType::kTH1F, {axisPhi}});                                                                    /// Σc0
+    registry.add("Data/hPhiSoftPiScPlusPlus", "#pi^{+} #leftarrow #Sigma_{c}^{++} candidates; #Phi ; counts;", {HistType::kTH1F, {axisPhi}});                                                            /// Σc++
+    registry.add("Data/hPhiSoftPiSc0PlusPlus", "#pi^{#pm} #leftarrow #Sigma_{c}^{0, ++} candidates; #Phi ; counts;", {HistType::kTH1F, {axisPhi}});                                                      /// Σc0,++
+
```

Comment:
```suggestion
    registry.add("Data/hDeltaMassSc0", "#Sigma_{c}^{0} candidates; #it{M}(K0sP #pi) - #it{M}(K0sP) (GeV/#it{c}^{2}); counts;", {HistType::kTH1D, {{200, 0.13, 0.23}}});                                                                  /// Σc0
    registry.add("Data/hDeltaMassScPlusPlus", "#Sigma_{c}^{++} candidates; #it{M}(K0sP#pi) - #it{M}(K0sP) (GeV/#it{c}^{2}); counts;", {HistType::kTH1D, {{200, 0.13, 0.23}}});                                                           /// Σc++
    registry.add("Data/hDeltaMassSc0PlusPlus", "#Sigma_{c}^{0, ++} candidates; #it{M}(K0sP#pi) - #it{M}(K0sP) (GeV/#it{c}^{2}); counts;", {HistType::kTH1D, {{200, 0.13, 0.23}}});                                                       /// Σc0,++
    registry.add("Data/hDeltaMassSc0VsPt", "#Sigma_{c}^{0} candidates; #it{M}(K0sP #pi) - #it{M}(K0sP) (GeV/#it{c}^{2}); #it{p}_{T}(#Sigma_{c}^{0}) (GeV/#it{c});", {HistType::kTH2D, {{200, 0.13, 0.23}, {36, 0., 36.}}});              /// Σc0
    registry.add("Data/hDeltaMassScPlusPlusVsPt", "#Sigma_{c}^{++} candidates; #it{M}(K0sP #pi) - #it{M}(K0sP) (GeV/#it{c}^{2}); #it{p}_{T}(#Sigma_{c}^{++}) (GeV/#it{c});", {HistType::kTH2D, {{200, 0.13, 0.23}, {36, 0., 36.}}});     /// Σc++
    registry.add("Data/hDeltaMassSc0PlusPlusVsPt", "#Sigma_{c}^{0, ++} candidates; #it{M}(K0sP #pi) - #it{M}(K0sP) (GeV/#it{c}^{2}); #it{p}_{T}(#Sigma_{c}^{++}) (GeV/#it{c});", {HistType::kTH2D, {{200, 0.13, 0.23}, {36, 0., 36.}}}); /// Σc0, ++
    registry.add("Data/hEtaSc0", "#Sigma_{c}^{0} candidates; #eta ; counts;", {HistType::kTH1D, {axisEta}});                                                                                                                             /// Σc0
    registry.add("Data/hEtaScPlusPlus", "#Sigma_{c}^{++} candidates; #eta ; counts;", {HistType::kTH1D, {axisEta}});                                                                                                                     /// Σc++
    registry.add("Data/hEtaSc0PlusPlus", "#Sigma_{c}^{0, ++} candidates; #eta ; counts;", {HistType::kTH1D, {axisEta}});                                                                                                                 /// Σc0,++
    registry.add("Data/hEtaSc0PlusPlusVsPt", "#Sigma_{c}^{0, ++} candidates; #eta; #it{p}_{T}(#Sigma_{c}^{0}) (GeV/#it{c});", {HistType::kTH2D, {axisEta, axisPt}});                                                                     /// Σc0,++
    registry.add("Data/hYSc0PlusPlusVsPt", "#Sigma_{c}^{0, ++} candidates; y (rapidity); #it{p}_{T}(#Sigma_{c}^{0}) (GeV/#it{c});", {HistType::kTH2D, {axisY, axisPt}});                                                                 /// Σc0,++
    registry.add("Data/hPhiSc0", "#Sigma_{c}^{0} candidates; #Phi ; counts;", {HistType::kTH1D, {axisPhi}});                                                                                                                             /// Σc0
    registry.add("Data/hPhiScPlusPlus", "#Sigma_{c}^{++} candidates; #Phi ; counts;", {HistType::kTH1D, {axisPhi}});                                                                                                                     /// Σc++
    registry.add("Data/hPhiSc0PlusPlus", "#Sigma_{c}^{0, ++} candidates; #Phi ; counts;", {HistType::kTH1D, {axisPhi}});                                                                                                                 /// Σc0,++

    /// softPions
    registry.add("Data/hPtSoftPiSc0", "#pi^{-} #leftarrow #Sigma_{c}^{0} candidates; #it{p}_{T}(#pi^{-} #leftarrow #Sigma_{c}^{0}) (GeV/#it{c}); counts;", {HistType::kTH1D, {axisPt}});                 /// Σc0
    registry.add("Data/hPtSoftPiScPlusPlus", "#pi^{+} #leftarrow #Sigma_{c}^{++} candidates; #it{p}_{T}(#pi^{+} #leftarrow #Sigma_{c}^{0}) (GeV/#it{c}); counts;", {HistType::kTH1D, {axisPt}});         /// Σc++
    registry.add("Data/hPtSoftPiSc0PlusPlus", "#pi^{#pm} #leftarrow #Sigma_{c}^{0, ++} candidates; #it{p}_{T}(#pi^{#pm} #leftarrow #Sigma_{c}^{0}) (GeV/#it{c}); counts;", {HistType::kTH1D, {axisPt}}); /// Σc0,++
    registry.add("Data/hEtaSoftPiSc0", "#pi^{-} #leftarrow #Sigma_{c}^{0} candidates; #eta ; counts;", {HistType::kTH1D, {axisEta}});                                                                    /// Σc0
    registry.add("Data/hEtaSoftPiScPlusPlus", "#pi^{+} #leftarrow #Sigma_{c}^{++} candidates; #eta ; counts;", {HistType::kTH1D, {axisEta}});                                                            /// Σc++
    registry.add("Data/hEtaSoftPiSc0PlusPlus", "#pi^{#pm} #leftarrow #Sigma_{c}^{0, ++} candidates; #eta ; counts;", {HistType::kTH1D, {axisEta}});                                                      /// Σc0,++
    registry.add("Data/hPhiSoftPiSc0", "#pi^{-} #leftarrow #Sigma_{c}^{0} candidates; #Phi ; counts;", {HistType::kTH1D, {axisPhi}});                                                                    /// Σc0
    registry.add("Data/hPhiSoftPiScPlusPlus", "#pi^{+} #leftarrow #Sigma_{c}^{++} candidates; #Phi ; counts;", {HistType::kTH1D, {axisPhi}});                                                            /// Σc++
    registry.add("Data/hPhiSoftPiSc0PlusPlus", "#pi^{#pm} #leftarrow #Sigma_{c}^{0, ++} candidates; #Phi ; counts;", {HistType::kTH1D, {axisPhi}});                                                      /// Σc0,++

```

Do not use histograms with float precision (especially for `TH1`s), otherwise you risk to have unexpected roundings due to the floating point precision at pow(2, 24) - 1 = 16777215

Reply:
Thank you very much. I have changed to TH1D.